pkg/mkimage-raw-efi/install

```diff
@@ -155,25 +140,25 @@ INSTALL_PERSIST=`cat /proc/cmdline | tr ' ' '\012' | sed -ne '/^eve_persist_disk
 INSTALL_PERSIST=${INSTALL_PERSIST:-$INSTALL_DEV}
 
 # now lets figure out whether we have installation material
-mkdir /parts || :
 CONFIG_PART=$(find_part CONFIG "$(root_dev)")
 CONFIG_PART="${CONFIG_PART:+"/dev/"}${CONFIG_PART:-"/bits/config.img"}"
 if [ -e "$CONFIG_PART" ]; then
    dd if="$CONFIG_PART" of=/parts/config.img bs=1M
-   tr ' ' '\012' < /proc/cmdline | sed -ne '/^eve_install_server=/s#^.*=##p' > /tmp/eve_install_server
-   [ ! -s /tmp/eve_install_server ] || mcopy -i /parts/config.img -o /tmp/eve_install_server ::/server
+else
+   mkfs.vfat -v -n CONFIG -C /parts/config.img 1024
+   mcopy -i /parts/config.img -s /config/* ::/
 fi
+# the only thing we override in /config for now is server
+tr ' ' '\012' < /proc/cmdline | sed -ne '/^eve_install_server=/s#^.*=##p' > /parts/eve_install_server
+[ ! -s /parts/eve_install_server ] || mcopy -i /parts/config.img -o /parts/eve_install_server ::/server
 
-FILE_PARTS="persist.img config.img rootfs.img"
-for i in EFI boot; do
-  SOURCE="/bits/$i"
-  [ ! -e "$SOURCE" ] || FILE_PARTS="$FILE_PARTS $i"
-done
-for i in $FILE_PARTS; do
-   SOURCE="/bits/$i"
-   [ -e "$SOURCE" ] || SOURCE=/dev/null
-   [ -e "/parts/$i" ] || ln -s "$SOURCE" "/parts/$i"
-done
+# if there's something in /bits -- that's the ultimate source
+ln -s /bits/* /parts 2>/dev/null
+
+# and now a few measures of last resort
+[ -e /parts/rootfs.img ] || ln -s "/dev/$(mounted_dev /root)" /parts/rootfs.img
+[ -e /parts/EFI ] || ln -s /root/EFI /parts/EFI
+[ -e /parts/boot ] || ln -s /root/boot /parts/boot
 
 # finally lets see if we were given any overrides
 grep -q eve_install_skip_rootfs /proc/cmdline && trunc /parts/rootfs.img
```

Comment:
Please rename `eve_install_skip_config` to `eve_install_skip_persist` few lines below.

Reply:
good catch